segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ThrottlerCalculator.java

```diff
@@ -266,6 +268,52 @@ ThrottlerName getName() {
         }
     }
 
+
+    /**
+     * Calculates the amount of time to wait before adding more requests to the queue in order to relieve pressure
+     * from the DurableDataLog. This is based on static information from the DurableDataLog's {@link WriteSettings} and dynamic
+     * information from its {@link QueueStats}.
+     *
+     * Throttle is applied  when total outstanding bytes is more than set limit.
+     * The value of throttle is linearly proportional to percentage of excess outstanding bytes with respected to max limit
+     * Max throttle value is capped at maxDelayMillis.
+     *
+     */
+    private static class DurableDataLogOutstandingBytesThrottler extends Throttler {
+        private final int maxOutstandingBytes;
+        private final int baseDelay;
+        private final Supplier<QueueStats> getQueueStats;
+
+        DurableDataLogOutstandingBytesThrottler(@NonNull WriteSettings writeSettings, @NonNull Supplier<QueueStats> getQueueStats, int maxDelayMillis) {
+            this.maxOutstandingBytes = writeSettings.getMaxOutstandingBytes();
+            this.baseDelay = maxDelayMillis;
+            this.getQueueStats = getQueueStats;
+        }
+
+        @Override
+        boolean isThrottlingRequired() {
+            QueueStats stats = this.getQueueStats.get();
+            return stats.getTotalLength() > this.maxOutstandingBytes;
+        }
+
+        @Override
+        int getDelayMillis() {
+            if (isThrottlingRequired()) {
+                QueueStats stats = this.getQueueStats.get();
+                val excess = stats.getTotalLength() - this.maxOutstandingBytes;
+                return Math.min(this.baseDelay, (int) Math.ceil(1.0 * excess * this.baseDelay / this.maxOutstandingBytes));
```

Comment:
1.0 is not needed. Also have parentheses around (excess / max) for clarity and to preserve precision.